src/utils.c

```diff
@@ -1,342 +1,459 @@
-// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
-// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
-// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
-// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
-// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in
+//       the documentation and/or other materials provided with the
+//       distribution.
+
+//     * Neither the name of UChicago Argonne, LLC, Argonne National
+//       Laboratory, ANL, the U.S. Government, nor the names of its
+//       contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+
+// THIS SOFTWARE IS PROVIDED BY UChicago Argonne, LLC AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL UChicago
+// Argonne, LLC OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
 #include "utils.h"
+#include <stdint.h>
 
-//for windows build
+// for windows build
 #ifdef WIN32
-	#ifdef PY3K
-	void PyInit_libtomopy(void){}
-	#else
-	void initlibtomopy(void){}
-	#endif
+#    ifdef PY3K
+void
+PyInit_libtomopy(void)
+{
+}
+#    else
+void
+initlibtomopy(void)
+{
+}
+#    endif
 #endif
 
+//============================================================================//
 
-void 
-preprocessing(
-    int ry, int rz,
-    int num_pixels, float center, 
-    float *mov, float *gridx, float *gridy) 
+void
+preprocessing(int ry, int rz, int num_pixels, float center, float* mov,
+              float* gridx, float* gridy)
 {
-    int i;
-
-    for(i=0; i<=ry; i++)
+    for(int i = 0; i <= ry; ++i)
     {
-        gridx[i] = -ry/2.+i;
+        gridx[i] = -ry * 0.5f + i;
     }
 
-    for(i=0; i<=rz; i++)
+    for(int i = 0; i <= rz; ++i)
     {
-        gridy[i] = -rz/2.+i;
+        gridy[i] = -rz * 0.5f + i;
     }
 
-    *mov = ((float)num_pixels-1)/2.0-center;
-    if(*mov-floor(*mov) < 0.01) {
-        *mov += 0.01;
+    *mov = ((float) num_pixels - 1) * 0.50f - center;
```

Comment:
There's a trailing zero here that's not necessary.

Reply:
Fair point about separate commit for formatting. Normally I would have but wasn't paying attention to that when I rebased bc I was more concerned about ensuring any the MPI stuff wasn't included. The only C code of import affected by the optimizations was the functions listed in utils.c. For the algorithms it was FP consistency